docs/developer/buddypress-database-diagram.md

```diff
@@ -0,0 +1,388 @@
+  - `meta_key`: Metadata key.
+  - `meta_value`: Metadata value (usually stored as serialized data).
+
+---
+
+## Friends Tables
+
+### wp_bp_friends
+
+This table manages the friendship connections between users.
+
+- **Columns:**
+  - `id`: Unique ID for each friendship connection.
+  - `initiator_user_id`: The user who sent the friend request.
+  - `friend_user_id`: The user who received the friend request.
+  - `is_confirmed`: Whether the friendship has been confirmed (1) or is pending (0).
+  - `is_limited`: Whether the friendship has any limitations.
+  - `date_created`: Timestamp of when the friendship was initiated.
+
+#### Example Query:
+
+```sql
+SELECT * FROM wp_bp_friends WHERE initiator_user_id = 1;
+```
+
+---
+
+## Groups Tables
+
+### wp_bp_groups
+
+This table stores the details of groups created in BuddyPress.
+
+- **Columns:**
+  - `id`: Unique ID for the group.
+  - `creator_id`: User ID of the group creator.
+  - `name`: Name of the group.
+  - `slug`: Unique slug for the group.
+  - `description`: Group description.
+  - `status`: Group visibility status (public, private, hidden).
+  - `parent_id`: Parent group ID (if applicable).
+  - `enable_forum`: Whether the group has forums enabled.
+  - `date_created`: When the group was created.
+
+### wp_bp_groups_groupmeta
+
+Stores metadata associated with groups.
+
+- **Columns:**
+  - `id`: Unique ID for group metadata.
+  - `group_id`: The ID of the associated group.
+  - `meta_key`: Metadata key.
+  - `meta_value`: Metadata value (usually stored as serialized data).
+
+### wp_bp_groups_members
+
+Manages membership of users in groups.
+
+- **Columns:**
+  - `id`: Unique membership ID.
+  - `group_id`: The ID of the group.
+  - `user_id`: The ID of the user.
+  - `inviter_id`: The ID of the user who invited the member (if applicable).
+  - `is_admin`: Whether the user is an admin of the group.
+  - `is_mod`: Whether the user is a moderator of the group.
+  - `user_title`: The role or title of the user within the group.
+  - `date_modified`: When the membership was last modified.
+  - `is_confirmed`: Whether the membership is confirmed.
+  - `is_banned`: Whether the user is banned from the group.
+  - `invite_sent`: Whether an invitation has been sent.
+
+#### Example Query:
+
+```sql
+SELECT * FROM wp_bp_groups_members WHERE group_id = 2;
+```
+
+---
+
+## Messaging Tables
+
+### wp_bp_messages_messages
+
+Stores private messages sent between users.
+
+- **Columns:**
+  - `id`: Unique message ID.
+  - `thread_id`: ID of the conversation thread.
+  - `sender_id`: User ID of the message sender.
+  - `subject`: Subject of the message.
+  - `message`: The actual message content.
+  - `date_sent`: When the message was sent.
+
+### wp_bp_messages_meta
+
+Stores metadata associated with messages.
+
+- **Columns:**
+  - `id`: Unique meta ID.
+  - `message_id`: The ID of the associated message.
+  - `meta_key`: Metadata key.
+  - `meta_value`: Metadata value.
+
+---
```

Comment:
It should be in `docs/developer/components/messages/db-schema.md` and use a table like the what I've written for the Activity above.

⚠️ the `wp_bp_messages_recipients` table is missing